File: solutions/php/0035-search-insert-position..php
<?php

class Solution
{
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer
     */
    public function searchInsert(array $nums, int $target): int
    {
        $index = count($nums);
        $left = 0;
        $right = $index - 1;

        while ($left <= $right) {
            $mid = intval(($left + $right) / 2);

            if ($target === $nums[$mid]) {
                return $mid;
            }

            if ($target < $nums[$mid]) {
                $index = $mid;
                $right = $mid - 1;
            } else {
                $left = $mid + 1;
            }
        }

        return $index;
    }
}

File: solutions/go/0035-search-insert-position.go
package main

func searchInsert(nums []int, target int) int {
	index := len(nums)
	left, right := 0, index-1

	for left <= right {
		mid := (left + right) / 2

		if target == nums[mid] {
			return mid
		}

		if target < nums[mid] {
			index = mid
			right = mid - 1
		} else {
			left = mid + 1
		}
	}

	return index
}
